Source/DafnyLanguageServer/MarkupTestFile.cs

```diff
@@ -0,0 +1,183 @@
+// Modified after being taken from https://github.com/dotnet/roslyn/blob/576abfa9a8f2d64967ceabe7fa0e5b7ebe550752/src/Compilers/Test/Core/MarkedSource/MarkupTestFile.cs
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Security.Policy;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Dafny.LanguageServer.Workspace;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.Dafny.LanguageServer {
+  /// <summary>
+  /// To aid with testing, we define a special type of text file that can encode additional
+  /// information in it.  This prevents a test writer from having to carry around multiple sources
+  /// of information that must be reconstituted.  For example, instead of having to keep around the
+  /// contents of a file *and* and the location of the cursor, the tester can just provide a
+  /// string with the "$" character in it.  This allows for easy creation of "FIT" tests where all
+  /// that needs to be provided are strings that encode every bit of state necessary in the string
+  /// itself.
+  /// 
+  /// The current set of encoded features we support are: 
+  /// 
+  /// $$ - The position in the file.  There can be at most one of these.
+  /// 
+  /// [| ... |] - A span of text in the file.  There can be many of these and they can be nested
+  /// and/or overlap the $ position.
+  /// 
+  /// {|Name: ... |} A span of text in the file annotated with an identifier.  There can be many of
+  /// these, including ones with the same name.
+  /// 
+  /// Additional encoded features can be added on a case by case basis.
+  /// </summary>
+  public static class MarkupTestFile {
+    private const string SpanStartString = "[>";
+    private const string SpanEndString = "<]";
+    private const string NamedSpanEndString = "<}";
+
+    private static void Parse(
+        string input, out string output, out List<int> positions, out IDictionary<string, List<TextSpan>> spans) {
+      positions = new List<int>();
+      var tempSpans = new Dictionary<string, List<TextSpan>>();
+
+      var outputBuilder = new StringBuilder();
+
+      // A stack of span starts along with their associated annotation name.  [||] spans simply
+      // have empty string for their annotation name.
+      var spanStartStack = new Stack<(int matchIndex, string name)>();
+      var namedSpanStartStack = new Stack<(int matchIndex, string name)>();
+
+      var r = new Regex(@"(?<Position>\>\<)|(?<SpanStart>\[\>)|(?<SpanEnd>\<\])|(?<NameSpanStart>\{\>([-_.A-Za-z0-9\+]+)\:)|(?<NameSpanEnd>\<\})");
```

Comment:
```suggestion
      var r = new Regex(@"(?<Position>><)|(?<SpanStart>\[>)|(?<SpanEnd><\])|(?<NameSpanStart>\{>([-_.A-Za-z0-9\+]+)\:)|(?<NameSpanEnd><\})");
```
My IDE makes a good job showing me what is a literal character and what is regex construct.